biorxiv/time_to_publication/time_to_published_refurbished.py

```diff
@@ -0,0 +1,240 @@
+
+
+# In[6]:
+
+
+kmf.fit(
+    preprints_w_published_dates["time_to_published"].dt.total_seconds() / 60 / 60 / 24,
+    event_observed= ~preprints_w_published_dates["published_doi"].isna(),
+)
+
+
+# In[7]:
+
+
+kmf.median_survival_time_
+
+
+# In[8]:
+
+
+overall_preprint_survival = (
+    kmf.survival_function_
+    .reset_index()
+    .assign(label="all_papers")
+)
+overall_preprint_survival.head()
+
+
+# In[9]:
+
+
+g = (
+    p9.ggplot(overall_preprint_survival, p9.aes(x="timeline", y="KM_estimate", color="label"))
+    + p9.geom_point()
+    + p9.ylim(0,1)
+    + p9.xlim(0, 2550)
+)
+print(g)
+
+
+# # Calculate Category Survival Function
+
+# This section measures how long it takes for certain categories to get preprints published.
+
+# In[10]:
+
+
+entire_preprint_df = pd.DataFrame([], columns=["timeline", "KM_estimate", "category"])
+half_life = []
+for cat, grouped_df in preprints_w_published_dates.groupby("category"):
+    temp_df = preprints_w_published_dates.query(f"category=='{cat}'")
+    kmf.fit(
+        temp_df["time_to_published"].dt.total_seconds() / 60 / 60 / 24,
+        event_observed= ~temp_df["published_doi"].isna(),
+    )
+    
+    half_life.append({
+        "category":cat, 
+        "half_life_time":kmf.median_survival_time_
+    })
+    
+    entire_preprint_df = (
+        entire_preprint_df
+        .append(
+            kmf.survival_function_
+            .reset_index()
+            .assign(category=cat)
+        )
+    )
+
+
+# In[11]:
+
+
+g = (
+    p9.ggplot(
+        entire_preprint_df, 
+        p9.aes(x="timeline", y="KM_estimate", color='category')
+    )
+    + p9.geom_line()
+    + p9.ylim(0,1)
+    + p9.xlim(0,2550)
+    + p9.labs(
+        x="timeline (days)",
+        y="proportion of unpublished biorxiv paper"
+    )
+)
+print(g)
+
+
+# In[12]:
+
+
+category_half_life = (
+    pd.DataFrame
+    .from_records(half_life)
+    .replace(np.inf, (temp_df["time_to_published"].dt.total_seconds() / 60 / 60 / 24).max())
+)
+category_half_life
+
+
+# In[13]:
+
+
+g = (
+    p9.ggplot(
+        category_half_life.query("category!='none'"),
+        p9.aes(x="category", y="half_life_time")
+    )
+    + p9.geom_col()
+    + p9.scale_x_discrete(
+        limits=(
+            category_half_life
+            .query("category!='none'")
+            .sort_values("half_life_time", ascending=False)
+            .category
+            .tolist()
+        )
+    )
+    + p9.coord_flip()
+    + p9.labs(
+        x="preprint category",
+        y="Days Until 50% of Preprints are Published",
+        title="Preprint Category Half-Life"
+    )
+)
+g.save("output/preprint_category_halflife.svg", dpi=500)
+g.save("output/preprint_category_halflife.png", dpi=500)
+print(g)
+
+
+# Take home Results:
+#     1. The average amount of time for half of all preprints to be published is 348 days (~1 year)
+#     2. Biophysics and biochemistry are two categories that take the least time to have half their preprints published
```

Comment:
I could see a short half life meaning either 'papers in this field don't take long to get published' or 'authors in this field submit a preprint once their manuscript is already accepted/close to it'. Is there a relationship between document distance and time to publish in particular fields?

Reply:
>Is there a relationship between document distance and time to publish in particular fields? 

Great question. This would be a great follow up to my overall doc distance to time published figure (publication_delay_experiment notebook). Will look into it, but hypothesis here is that some categories would have a lower slope than other categories if I were to perform linear regression for each individual category.